pyV2DL3/eventdisplay/fillRESPONSE.py

```diff
@@ -225,31 +225,64 @@ def fill_direction_migration(
     irf_interpolator.set_irf("hAngularLogDiffEmc_2D")
 
     rpsf_final = []
+    rpsf_test = []
 
     for offset in camera_offsets:
 
+        # direction diff (rad, energy),
         direction_diff, axis = irf_interpolator.interpolate([pedvar, zenith, offset])
 
+        # energy axis from ~ 0.1 - 100 TeV
+        direction_diff = direction_diff[:, 5:-10]
+        axis[0] = axis[0][5:-10]
         _, eLow, eHigh = bin_centers_to_edges(axis[0])
         rad_edges, rLow, rHigh = bin_centers_to_edges(axis[1])
 
-        # Normalize rpsf by solid angle
-        rad_width_deg = np.diff(np.power(10, rad_edges))
-        e_sum = np.sum(
-            direction_diff
-            * 2
-            * rad_width_deg[:, np.newaxis]
-            * np.pi
-            * np.power(10, axis[1])[:, np.newaxis],
-            axis=0,
-        )
-        normsum = np.divide(((180 / np.pi) ** 2), e_sum, where=e_sum != 0)
+        ### normalize by using rad**bins and norm to rad ** 2 bins
+        rad_edges, rLow, rHigh = bin_centers_to_edges(axis[1], logaxis=True)
 
-        rpsf = direction_diff * normsum
-        rpsf_final.append(np.nan_to_num(rpsf).cumsum(axis=0))
+        rad_width_deg = np.diff(np.power(10, rad_edges))
+        norm = np.sum(direction_diff * np.repeat(rad_width_deg[..., np.newaxis], 15, axis=1) \
+               / np.repeat(((rLow + rHigh) / 2)[..., np.newaxis], 15, axis=1), axis=0)
+        norm = norm * 2 * np.pi
+        direction_diff = direction_diff / (np.repeat(((rLow + rHigh) / 2)[..., np.newaxis], 15, axis=1) ** 2)
+        normed = direction_diff / norm * ((180 / np.pi) ** 2)
+        rpsf = normed
```

Comment:
Why do you need here the variable normed or rpsf. Couldn't you use only normed?

Reply:
fixed 965d8a3dda653c515d31324940f4325f727501e2